site/docs/security/tls.md

```diff
@@ -0,0 +1,178 @@
+certificates that clients keys were signed by.
+
+    keytool -keystore client.truststore.jks -alias CARoot -import -file ca-cert
+
+In contrast to the keystore, which stores each machine’s own identity, the truststore of a client stores all the certificates
+that the client should trust. Importing a certificate into one’s truststore also means trusting all certificates that are signed
+by that certificate. As the analogy above, trusting the government (CA) also means trusting all passports (certificates) that
+it has issued. This attribute is called the chain of trust, and it is particularly useful when deploying TLS on a large BookKeeper cluster.
+You can sign all certificates in the cluster with a single CA, and have all machines share the same truststore that trusts the CA.
+That way all machines can authenticate all other machines.
+
+## Signing the certificate
+
+The next step is to sign all certificates in the keystore with the CA we generated. First, you need to export the certificate from the keystore:
+
+    keytool -keystore bookie.keystore.jks -alias localhost -certreq -file cert-file
+
+Then sign it with the CA:
+
+    openssl x509 -req -CA ca-cert -CAkey ca-key -in cert-file -out cert-signed -days {validity} -CAcreateserial -passin pass:{ca-password}
+
+Finally, you need to import both the certificate of the CA and the signed certificate into the keystore:
+
+    keytool -keystore bookie.keystore.jks -alias CARoot -import -file ca-cert
+    keytool -keystore bookie.keystore.jks -alias localhost -import -file cert-signed
+
+The definitions of the parameters are the following:
+
+1. `keystore`: the location of the keystore
+2. `ca-cert`: the certificate of the CA
+3. `ca-key`: the private key of the CA
+4. `ca-password`: the passphrase of the CA
+5. `cert-file`: the exported, unsigned certificate of the bookie
+6. `cert-signed`: the signed certificate of the bookie
+
+(TBD: add a script to automatically generate truststores and keystores.)
+
+## Configuring Bookies
+
+Bookies support TLS for connections on the same service port. In order to enable TLS, you need to configure `tlsProvider` to be either
+`JDK` or `OpenSSL`. If `OpenSSL` is configured but the open ssl library is not found, it will automatically fall back to use default
+JDK based TLS provider.
+
+The following TLS configs are needed on the bookie side:
+
+    tlsProvider=OpenSSL
+    # key store
+    tlsKeyStoreType=JKS
+    tlsKeyStore=/var/private/tls/bookie.keystore.jks
+    tlsKeyStorePasswordPath=/var/private/tls/bookie.keystore.passwd
+    # trust store
+    tlsTrustStoreType=JKS
+    tlsTrustStore=/var/private/tls/bookie.truststore.jks
+    tlsTrustStorePasswordPath=/var/private/tls/bookie.truststore.passwd
+
+NOTE: it is important to restrict access to the store files and corresponding password files via filesystem permissions.
+
+Optional settings that are worth considering:
+
+1. tlsClientAuthentication=false: Enable/Disable using TLS for authentication.
```

Comment:
This config when enabled will authenticate the other end of the communication channel. It should be enabled on both bookie and client for MutualTLS.